src/node/delay.rs

```diff
@@ -432,7 +435,9 @@ impl AudioProcessor for DelayReader {
         let quantum_duration = RENDER_QUANTUM_SIZE as f64 * dt;
 
         let delay_param = params.get(&self.delay_time);
-        let in_cycle = self.in_cycle.load(Ordering::SeqCst);
+
+        let latest_frame_written = self.latest_frame_written.load(Ordering::SeqCst);
+        let in_cycle = latest_frame_written != scope.current_frame;
```

Comment:
I'm not that sure this is right when in cycle actually:
- if not in cycle the order of rendering is guaranteed, therefore we can trust this check
- but when in cycle, the order of rendering is not guaranteed (at least it wasn't, did it changed somehow?), so the check could fail sometime?

Reply:
Hmmmm right, it is quite nuanced actually and my code needs to be improved
- When not in a cycle, there is a edge from writer to reader so writer always renders first
- When in a cycle, that edge is dropped. The nature of the cycle means output of the reader feeds into the writer, hence, the reader always renders first
- However, when the user breaks the cycle manually by dropping some other connections, the order is no longer guaranteed and thus random

The last case is a bit of an edge case of course, but I could make the in_cycle property sticky (one you are in a cycle, treat as if you will always stay in that cycle) so we prevent erratic behaviour in rendering of sub-quantum delays.

Another issue I realize is when the Writer is dropped, it will stop updating the `latest_frame_written`. This causes the Reader to set `in_cycle` true, disallowing sub-quantum delays. If the delay is set to 100 samples this behaviour is not right becase the final 28 frames om the last Writer call should still render directly in the first 28 frames of the next Reader output.

Pfft, not simple stuff, I will iterate again